lib/iris/cube.py

```diff
@@ -3131,6 +3084,10 @@ def __eq__(self, other):
         if isinstance(other, Cube):
             result = self.metadata == other.metadata
 
+            # check the cube fill-value.
+            if result:
+                result = self.fill_value == other.fill_value
```

Comment:
I reasoned that previously we didn't account for `dtype` when performing equality, so we shouldn't do so here explicitly. However, we do need to consider `fill_value`, hence why I only added this.